rosbot_gazebo/test/simulation_test_node.py

```diff
@@ -97,6 +102,9 @@ def timer_callback(self):
         self.publish_cmd_vel_messages()
         self.lookup_transform_odom()
 
+    def scan_callback(self, data: LaserScan):
+        self.scan_event.set()
```

Comment:
Make sure message is not empty
```suggestion
        if data.ranges:
            self.scan_event.set()
```